plugins/jobtype/src/azkaban/jobtype/HadoopSecureSparkWrapper.java

```diff
@@ -0,0 +1,157 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package azkaban.jobtype;
+
+import static org.apache.hadoop.security.UserGroupInformation.HADOOP_TOKEN_FILE_LOCATION;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.security.PrivilegedExceptionAction;
+import java.util.Arrays;
+import java.util.Properties;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.security.UserGroupInformation;
+import org.apache.hadoop.security.token.Token;
+import org.apache.log4j.Logger;
+
+import azkaban.jobExecutor.ProcessJob;
+import azkaban.security.commons.HadoopSecurityManager;
+import azkaban.utils.Props;
+
+public class HadoopSecureSparkWrapper {
+
+  // param values
+  private static boolean securityEnabled;
+
+  private static final Logger logger = Logger.getRootLogger();
+
+  /**
+   * Entry point: a Java wrapper to the spark-submit command
+   * 
+   * @param args
+   * @throws Exception
+   */
+  public static void main(final String[] args) throws Exception {
+
+    Properties jobProps = loadCombinedProps();
+    HadoopConfigurationInjector.injectResources(new Props(null, jobProps));
+
+    // set up hadoop related configurations
+    final Configuration conf = new Configuration();
+    UserGroupInformation.setConfiguration(conf);
+    securityEnabled = UserGroupInformation.isSecurityEnabled();
+
+    if (shouldProxy(jobProps)) {
+      UserGroupInformation proxyUser = null;
+      String userToProxy = jobProps.getProperty("user.to.proxy");
+      if (securityEnabled) {
+        proxyUser = createSecurityEnabledProxyUser(userToProxy);
+      } else {
+        proxyUser = UserGroupInformation.createRemoteUser(userToProxy);
+      }
+
+      logger.info("Proxied as user " + userToProxy);
+
+      proxyUser.doAs(new PrivilegedExceptionAction<Void>() {
+        @Override
+        public Void run() throws Exception {
+          runSpark(args);
+          return null;
+        }
+      });
+
+    } else {
+      logger.info("Not proxying. ");
+      runSpark(args);
+    }
+  }
+
+  /**
+   * Actually calling the spark-submit command
+   * 
+   * @param args
+   */
+  private static void runSpark(String[] args) {
+
+    logger.info("args: " + Arrays.toString(args));
+
+    org.apache.spark.deploy.SparkSubmit$.MODULE$.main(args);
+  }
+
+  /**
+   * Perform all the magic required to get the proxyUser in a securitized grid
+   * 
+   * @param userToProxy
+   * @return
+   * @throws IOException
+   */
+  private static UserGroupInformation createSecurityEnabledProxyUser(String userToProxy)
```

Comment:
Hadoop[Pig | Hive | Spark]SecureWrapper are all very much alike.
Refactoring wise, I think they should all be grouped together.
Will work on it